### DISCUSSION TITLE
Add Creation and completion dates

### DISCUSSION THREAD

**Discussion by FranciscoTrigo** [jotaen/xit#59] [File Format]
This looks very cool! I currently use todo.txt and im thinking to change to this.
Maybe the format can have optional creation and completion dates for items?

A way that I think this can be done is in the first line of a long description (or the second line of the item) like this:

```
[ ] Paint the room
Created: 2023-02-01 Completed: 2023-03-04
More description if needed
```

It would probably be neater to choose some kind of symbol or abbreviation to denote Created and Completed instead of using the words.

**Comment by jotaen** [comment]
One way to kind of make that happen right now would be by using tags, and establishing your own convention, such as:

```
[ ] Paint the room #created=2023-02-01 #completed=2023-03-04
[ ] Do something
    #created=2023-02-01 #done=2023-03-04
```

That would also allow for tooling support, along the lines of “give me all items with tag #created where the tag value is smaller than 2023-02-15”.

**Comment by FranciscoTrigo** [comment]
Oh that actually sounds pretty neat, I honestly did not understand the use case for tags with values before this, very nice.

**Reply by jotaen** [reply]
Yeah, good point about the use-cases! I’ve added some demo usages to [the project homepage](https://xit.jotaen.net/), I hope that helps others to get a better idea.

<img width="765" alt="Screenshot 2023-04-30 at 09 42 47" src="https://user-images.githubusercontent.com/3618384/235341777-92c0a80a-555a-4820-ab9d-ca2f667d6cdc.png">
